lint/src/main/java/permissions/dispatcher/NoDelegateOnResumeDetector.java

```diff
@@ -0,0 +1,128 @@
+package permissions.dispatcher;
+
+import com.android.tools.lint.client.api.UElementHandler;
+import com.android.tools.lint.detector.api.Category;
+import com.android.tools.lint.detector.api.Detector;
+import com.android.tools.lint.detector.api.Implementation;
+import com.android.tools.lint.detector.api.Issue;
+import com.android.tools.lint.detector.api.JavaContext;
+import com.android.tools.lint.detector.api.Scope;
+import com.android.tools.lint.detector.api.Severity;
+import com.intellij.psi.PsiType;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.uast.UCallExpression;
+import org.jetbrains.uast.UClass;
+import org.jetbrains.uast.UElement;
+import org.jetbrains.uast.UExpression;
+import org.jetbrains.uast.UIdentifier;
+import org.jetbrains.uast.UMethod;
+import org.jetbrains.uast.visitor.AbstractUastVisitor;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+public class NoDelegateOnResumeDetector extends Detector implements Detector.UastScanner {
+    static final Issue ISSUE = Issue.create("NoDelegateOnResumeDetector",
+            "Asking permissions at onResume will cause an infinite loop.",
+            "Asking permissions at onResume will cause an infinite loop when user selects \"never ask again\" and denies the permission. This is because onResume method is called after returning from a permission request check. This problem can be solved by moving the method call to onStart.",
```

Comment:
@mannodermaus Could you check my English?

Reply:
Looks generally good to me! Some tiny adjustments that I'd like to propose, though:

```
Asking permissions in onResume will cause an infinite loop when user selects \"Never ask again\" and denies the permission, since onResume is called again after returning from a permission request check. This problem can be solved by moving this method call to onStart.
```

Also, do you know if there is additional formatting support for Lint messages? Would be cool to wrap the function names in the comment inside \<code> brackets, and have them rendered `like this` in the doc.

Reply:
Also, in the call site below, you attach the method brackets to the description (i.e. `onResume()`). Would it make sense to add them here as well, for consistency?

Reply:
For formatting,

Here is the line that should be format lint message.

https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-api/src/main/java/com/android/tools/lint/detector/api/Issue.java#145

And RAW is this

http://static.javadoc.io/com.android.tools.lint/lint-api/25.3.0/com/android/tools/lint/detector/api/TextFormat.html#RAW

So, I put `` in order to show the method. Now the result is

![2018-06-10 21 39 45](https://user-images.githubusercontent.com/3675458/41201688-5774234a-6cf7-11e8-8eab-9d0c8374ac48.png)

However, Intellij seems not supporting the format...

![2018-06-10 21 40 08](https://user-images.githubusercontent.com/3675458/41201690-650a8c42-6cf7-11e8-9a31-b687e9c71f7f.png)

And the lint test is also ignore the format. This is why I didn't change the test casess

Reply:
Seems like the IDE is still rendering it in a pleasant fashion, though, so not having the monospace in the IntelliJ hint is acceptable imo. I think I prefer it this way!